contrib/init/upstart/docker.conf

```diff
@@ -70,3 +70,10 @@ post-start script
 		echo "$DOCKER_SOCKET is up"
 	fi
 end script
+
+pre-stop script
+        SHUTDOWN_RUNLEVELS='06'
+        if runlevel | cut -d' ' -f2 | grep -q [$SHUTDOWN_RUNLEVELS]; then
+                docker stop -t 30 $(docker ps -q)
```

Comment:
I should probably also put some logic in here to allow overriding of `-t` by sourcing `/etc/default/docker`?